react/modules/graphiql.md

```diff
@@ -0,0 +1,152 @@
+- the demo version of the application should be deployed on `gh-pages`, `netlify`, `heroku`, or another similar service.
+- after completing the task, create a pull request from **develop** to **main** branch. **Merging a Pull Request is not required**
+- describe in PR how to run your backend locally if you didn't deploy backend.
+- you should make your repo public after deadline
+Gitflow, workflow
+You are free to choose your own way of developing the project and the repository. Here are links to a description and analysis of popular practices.
+
+[Patterns for Managing Source Code Branches](https://martinfowler.com/articles/branching-patterns.html)
+[Gitflow Workflow](https://www.atlassian.com/ru/git/tutorials/comparing-workflows/gitflow-workflow)
+[GitHub flow](https://docs.github.com/en/get-started/quickstart/github-flow)
+[ThreeFlow](https://habr.com/ru/company/infopulse/blog/345826/)
+[GitLab Flow VS Git flow VS GitHub flow](https://yapro.ru/article/6172)
+## Technical requirements
+
+- localization (at least 2 languages). You should be able to change the language by clicking on the toggler/select in header.
+- semantic layout
+- the application should work at least in the latest version of the Google Chrome browser
+- you must use React 18 + **suspense** feature.(Gatsby, NextJS, Remix, etc. are not prohibited)
+- Private routes, 404, error boundary
+- Using jQuery in the main application code is not allowed.
+- You can use create-react-app, vite, css frameworks, any js/components libs you want, html and css pre-processors
+- You MUST use Typescript
+- ❗ it is forbidden to copy the code of other students or [GrahiQL](https://www.npmjs.com/package/@graphiql/react). This ban applies to html, css, and js code. You can use small code snippets from Stack Overflow, and other self-found sources on the Internet, except github repositories of course students.
+
+## Application design requirements
+
+- you are not limited in creativity but limited by the user's capabilities
+- the application quality is characterized by the elaboration of details, attention to typography (no more than three fonts per page, font size of at least 14 px, optimal [font and background contrast](https://snook.ca/technical/colour_contrast/colour.html)), carefully selected content
+- adaptive layout. The minimum page width of the application is 320px
+- interactivity of elements users can interact with. Element hover effects. Usage of different styles for the active and inactive state of the element. Smooth animations
+- the unity of styles of all pages of the application - the same fonts, button styles, indents, the same elements on all pages of the application have the same appearance and layout. Item colors and background images may vary. In this case, colors should be from the same palette, and background images from the same collection.
+
+## Description of function blocks
+
+### Welcome page(route)
+
```

Comment:
Did you forget to add a description here? Because I see it in the **Cross-check criteria** section

Reply:
No, I'm waiting for suggestions what we could add to the welcome page